charts/operator-wandb/charts/app/templates/_helpers.tpl

```diff
@@ -123,11 +123,13 @@ app deployments.
 {{- $bucket = printf "gs://%s/%s" $bucketValues.name (default "" $bucketValues.path) -}}
 {{- end -}}
 {{- if eq $bucketValues.provider "s3" -}}
-{{- if and $bucketValues.accessKey $bucketValues.secretKey -}}
-{{- $bucket = printf "s3://%s:%s@%s/%s" $bucketValues.accessKey $bucketValues.secretKey $bucketValues.name (default "" $bucketValues.path) -}}
-{{- else -}}
-{{- $bucket = printf "s3://%s/%s" $bucketValues.name (default "" $bucketValues.path) -}}
+{{- $credentials := "" -}}
+{{- if and (not $bucketValues.accessKey) (not $bucketValues.secretKey) $bucketValues.secretName -}}
+{{- $credentials = printf "$(ACCESS_KEY):$(SECRET_KEY)@" -}}
+{{- else if and $bucketValues.accessKey $bucketValues.secretKey -}}
+{{- $credentials = printf "%s:%s@" $bucketValues.accessKey $bucketValues.secretKey -}}
 {{- end -}}
+{{- $bucket = printf "s3://%s%s/%s" $credentials $bucketValues.name (default "" $bucketValues.path) -}}
```

Comment:
This is only required if the user configures the access and secret keys or the Secret name.
There's a possibility users rely on IAM, so they will not need to inform the access/secret pair or the secret name.
I think tweaking the ternary operator should get this done.

Reply:
@flamarion 
If the user doesn’t pass any of those and is relying on IAM
In this case, should bucket secret be created or not?
If not - what output are we expecting in "AZURE_STORAGE_KEY" ?

Reply:
The `AZURE_STORAGE_KEY` env var is required only when the schema is `az://....` 
For `s3://`, the user can or cannot use the secret pair. since it can use IAM in case they are in AWS